### PR TITLE
edge set refactor: separate filter and accessor 

### PR DESCRIPTION
**Pull request by yingjieMiao**
@ymatsuda 
1. Nothing really changed. 
2. Naming is really hard. May change some of the names later.

### CODE REVIEW

File: kifi-backend/modules/search/test/com/keepit/search/graph/EdgeSetTest.scala [hunk 1]
```diff
@@ -30,16 +30,16 @@ class EdgeSetTest extends Specification {
...
+//      edgeSet.filterByTimeRange(101, 119).destIdLongSet === Set[Long](4, 7, 8, 9)
+//
+//      // unsorted ids (reversed arrays)
+//      idArray = idArray.reverse
+//      timeArray = timeArray.reverse
+//      edgeSet = new TestEdgeSet(Id[Vertex](0), LongArraySet.from(idArray), timeArray)
+//
+//      edgeSet.filterByTimeRange(100, 130).destIdLongSet === Set[Long](1, 4, 5, 6, 7, 8, 9)
+//      edgeSet.filterByTimeRange(101, 119).destIdLongSet === Set[Long](4, 7, 8, 9)
     }
```
Comment (yingjieMiao) [inline kifi-backend/modules/search/test/com/keepit/search/graph/EdgeSetTest.scala:43]: The generic "filterByTimeRange" function is removed. I don't see it used anywhere. So this test is not useful.

File: kifi-backend/modules/search/app/com/keepit/search/graph/EdgeSetAccessor.scala [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  def seek(id: Id[D]): Boolean = seek(id.id)
+  def seek(id: Long): Boolean = {
+    _destId = id
+    destIdLongSet.contains(id)
+  }
+
+  def destId: Long = _destId
+}
+
+trait LongArrayBasedEdgeInfoAccessor[S, D] extends EdgeSetAccessor[S, D] {
```
Comment (yingjieMiao) [inline kifi-backend/modules/search/app/com/keepit/search/graph/EdgeSetAccessor.scala:22]: these apis were used in two classes in the old code.

File: kifi-backend/modules/search/test/com/keepit/search/graph/EdgeSetTest.scala (hunk 1 above)
Comment (ymatsuda) [inline kifi-backend/modules/search/test/com/keepit/search/graph/EdgeSetTest.scala:43]: used in SocialGraphInfo?

Comment (yingjieMiao) [inline kifi-backend/modules/search/test/com/keepit/search/graph/EdgeSetTest.scala:43]: They all use the LuceneBacked ones. Works in tests and dev mode.

Comment (yingjieMiao) [inline kifi-backend/modules/search/test/com/keepit/search/graph/EdgeSetTest.scala:43]: going to add a LuceneTimeRangeFilter test. Remove this one.

**Pull request merged**